LethalAPI.Core/CorePlugin.cs

```diff
@@ -12,50 +12,49 @@ namespace LethalAPI.Core;
 #pragma warning disable SA1401 // field should be made private
 #pragma warning disable SA1309 // Names should not start with an underscore. ie: _Logger.
 using System;
-using System.Diagnostics;
-using System.Reflection;
 
-using BepInEx;
-using BepInEx.Logging;
+using Features;
 using HarmonyLib;
 using MEC;
-using MonoMod.RuntimeDetour;
 
 /// <inheritdoc />
-[BepInPlugin(PluginInfo.PLUGIN_GUID, PluginInfo.PLUGIN_NAME, PluginInfo.PLUGIN_VERSION)]
-public class Plugin : BaseUnityPlugin
+public class CorePlugin : Plugin<CoreConfig>
 {
     /// <summary>
-    /// Gets the instance for the main api.
+    /// Gets the main instance of the core plugin.
     /// </summary>
-    public static Plugin Instance = null!;
-
-    /// <summary>
-    /// Gets the <see cref="BepInEx.Logging.Logger"/>.
-    /// </summary>
-    /// <summary>
-    /// The base logger.
-    /// </summary>
-    internal static ManualLogSource _Logger = null!;
+    public static CorePlugin Instance = null!;
 
     /// <summary>
     /// The harmony instance.
     /// </summary>
     internal static Harmony Harmony = null!;
 
-    private void Awake()
+    /// <inheritdoc />
+    // sets this so the config name isn't a mess. :)
+    public override string Name => "LethalApi-Core";
```

Comment:
2 reasons:
A) I figured that it was autogenerated since its in the obj. To prevent typeloadexceptions when bepinex isn't present, I'm trying to shift away from using autogenerated features and using statements. This makes it easier to troubleshoot when random typeloads are thrown (which is pretty safe at this point but regardless)
B) The logging scheme splits up the bootstrapper, loader, events and core. This helps speed up troubleshooting, but because the logging system utilizes the plugin name, it's easiest to just change the name of the plugin. We can change it back, but since it's instance based and Bepinex and Melon don't actually load the core, it's not gonna really make much of a difference.
IE: `LethalAPI-Bootstrapper` `LethalAPI-Core` `LethalAPI-Loader`

Reply:
That's fair. We should be consistent and not use any BepInEx PluginInfo then, though. (one of the fields used it iirc).